packages/calcite-components/src/components/tab-nav/tab-nav.tsx

```diff
@@ -323,6 +381,58 @@ export class TabNav {
   //
   //--------------------------------------------------------------------------
 
+  private findVisibleTabTitleIndex = (tabTitles: NodeListOf<Element>, isNext: boolean): number => {
+    const mobilePageWidth = this.el.getBoundingClientRect().width;
+    let visibleTabTitleIndex = -1;
+
+    for (let i = 0; i < tabTitles.length; i++) {
+      const tabTitle = tabTitles[i];
+      const tabTitleRect = tabTitle.getBoundingClientRect();
```

Comment:
note: If we can store `getBoundingClinetRect` anywhere and not call it unless necessary that would be best as it is an expensive dom query.